pkg/delete/loader.go

```diff
@@ -137,99 +120,101 @@ func parseDeleteFileDefinition(ctx *loaderContext, definition persistence.FileDe
 	return result, nil
 }
 
-func parseDeleteEntry(ctx *loaderContext, entry any) (pointer.DeletePointer, error) {
-
-	ptr, err := parseFullEntry(ctx, entry)
-
-	if str, ok := entry.(string); ok && err != nil {
-		return parseSimpleEntry(str)
+func convertToDeletePointer(entry any) (pointer.DeletePointer, error) {
+	if str, ok := entry.(string); ok {
+		return convertLegacy(str)
 	}
-
-	return ptr, err
+	return convert(entry)
 }
 
-func parseFullEntry(ctx *loaderContext, entry interface{}) (pointer.DeletePointer, error) {
-
+func convert(entry interface{}) (pointer.DeletePointer, error) {
 	var parsed persistence.DeleteEntry
-	err := mapstructure.Decode(entry, &parsed)
-	if err != nil {
+	if err := mapstructure.Decode(entry, &parsed); err != nil {
 		return pointer.DeletePointer{}, err
 	}
 
-	if a, known := ctx.knownApis[parsed.Type]; known {
-		p, err := parseAPIEntry(parsed, a)
-		if err != nil {
+	if a, known := api.NewAPIs()[parsed.Type]; known {
```

Comment:
Do we need to check the presence of `parsed.Type` first? I think the field is required.